pkg/k8s/watchers/watcher.go

```diff
@@ -461,16 +461,24 @@ func (k *K8sWatcher) InitK8sSubsystem(ctx context.Context, cachesSynced chan str
 
 	go func() {
 		log.Info("Waiting until local node addressing before starting watchers depending on it")
-		k.nodeDiscoverManager.WaitForLocalNodeInit()
-		if err := k.enableK8sWatchers(ctx, afterNodeInitResources); err != nil {
+		err := k.nodeDiscoverManager.WaitForIngressIPs(ctx)
+		if err != nil {
+			if !errors.Is(err, context.Canceled) {
+				log.WithError(err).Fatal("Ingress IPs not configured in time, was cilium-ingress-drone pod deployed?")
```

Comment:
Can we make these assumptions with this model?

I hadn't thought too far about lifecycle yet, but I'd presumed that we should handle the possibility that ingress is not deployed (in which case you don't configure ingress on that node / agent), and also handle cases where the pod gets deleted or redeployed, and in each of these cases you basically handle these events fresh and plumb the whole ingress piece.